scipy/interpolate/_inverse_distance_weighted.py

```diff
@@ -0,0 +1,298 @@
+import numpy as np
+from scipy.spatial import cKDTree, distance
+from scipy.interpolate.interpnd import NDInterpolatorBase, _ndim_coords_from_arrays
+
+__all__ = ['InverseDistanceWeightedNDInterpolator']
+
+class InverseDistanceWeightedNDInterpolator(NDInterpolatorBase):
+    """
+    Inverse Distance Weighted (IDW) interpolation.
+
+    Parameters
+    ----------
+    x : (npoints, ndims) ndarray of floats
+        Data point coordinates.
+    y : (npoints, ) ndarray of floats
+        Data values associated with the points.
+    rescale : bool, optional
+        Rescale points to a unit cube before performing interpolation.
+        This is useful if some of the input dimensions have incommensurable units
+        and differ by many orders of magnitude. Default is False.
+    fill_value : float or nan, optional
+        Value used to fill in for requested points outside of the convex hull
+        of the input points. Default is nan.
+    tree_options : dict, optional
+        Options passed to the underlying `cKDTree`.
+    local : bool, optional
+        If True, perform local interpolation, else global. Default is True.
+
+    Methods
+    -------
+    __call__(xi, p=2, k=3, distance_upper_bound=np.inf, **query_options)
+        Evaluate the interpolator at given points.
+
+    Examples
+    --------
+    >>> points = np.array([[0, 0], [1, 1], [2, 2]])
+    >>> values = np.array([0, 1, 2])
+    >>> interpolator = InverseDistanceWeightedNDInterpolator(points, values)
+    >>> interpolator([1.5, 1.5])
+    array([1.42105263])
+    >>> def exp_weight(dists, power, decorrelation_length=1.5):
+    ...     return np.exp(-(dists / decorrelation_length) ** p)
+    >>> interpolator([1.5, 1.5], weight_func=exp_weight)
+    array([1.38311737])
+
+    Notes
+    -----
+    The interpolation uses an inverse distance weighting method, where the
+    influence of each data point on the interpolated values is inversely
+    proportional to the distance from the point to a power `p` or specified
+    by the user.
+    """
+
+    def __init__(
+        self, x, y, rescale=False, fill_value=np.nan, tree_options=None, local=True
+    ):
+        NDInterpolatorBase.__init__(
+            self, x, y, rescale=rescale, need_contiguous=False, need_values=False
+        )
+        if tree_options is None:
+            tree_options = dict()
+        self.tree = cKDTree(self.points, **tree_options)
```

Comment:
Perhaps slightly cleaner and more consistent will __call__? WDYT?
```suggestion
        self, x, y, rescale=False, fill_value=np.nan, local=True, **tree_options
    ):
        NDInterpolatorBase.__init__(
            self, x, y, rescale=rescale, need_contiguous=False, need_values=False
        )
        self.tree = cKDTree(self.points, **tree_options)
```